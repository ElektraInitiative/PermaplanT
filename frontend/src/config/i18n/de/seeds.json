{
  "view_seeds": {
    "title": "Meine Samen",
    "btn_new_entry": "Neuer Eintrag",
    "search_placeholder": "Suche Samen...",
    "search_hint": "Derzeit wird nur die Samenbezeichnung bei der Suche berücksichtigt.",
    "fetching_error": "Sorry, ich habe Schwierigkeiten Samen von meinem Server zu erhalten. Bitte versuche es später erneut.",
    "fetching_single_error": "Sorry, ich konnte den Samen nicht von meinem Server laden. Möglicherweise existiert er nicht."
  },
  "delete_seed": {
    "btn_delete_seed": "Samen löschen"
  },
  "create_seed": {
    "title": "Samen Eintrag",
    "changes_model_title": "Eintrag abbrechen?",
    "changes_model_message": "Änderungen, die Sie vorgenommen haben, werden nicht gespeichert. Wollen Sie wirklich abbrechen?"
  },
  "create_seed_form": {
    "placeholder_binomial_name": "Tomate",
    "placeholder_origin": "Daheim",
    "placeholder_taste": "nussig",
    "placeholder_yield": "Hoch",
    "placeholder_price": "2,99€",
    "placeholder_generation": "0",
    "error_price_must_be_number": "Der Preis muss eine Zahl sein. z.B. 2,99",
    "btn_create_seed": "Samen speichern",
    "error_create_seed": "Sorry, ich habe Schwierigkeiten den Samen auf meinem Server abzulegen. Bitte versuche es später erneut.",
    "error_delete_seed": "Sorry, ich habe Schwierigkeiten den Samen vom Server zu löschen. Bitte versuche es später erneut.",
    "error_fetching_seeds": "Sorry, ich habe Schwierigkeiten Samen von meinem Server zu erhalten. Bitte versuche es später erneut.",
    "error_fetching_plants": "Sorry, ich habe Schwierigkeiten Pflanzen von meinem Server zu erhalten. Bitte versuche es später erneut.",
    "success": "Samen erfolgreich erstellt!"
  },
  "edit_seed_form": {
    "btn_edit_seed": "Samen editieren",
    "success": "Samen erfolgreich editiert!"
  },
  "harvest_year": "Bezugsjahr",
  "additional_name": "Zusätzlicher Name",
  "binomial_name": "Name der Pflanze",
  "quantity": "Menge",
  "origin": "Herkunft",
  "use_by": "Verbrauch bis",
  "quality": "Qualität",
  "taste": "Geschmack",
  "yield": "Ertrag",
  "price": "Preis",
  "generation": "Generation",
  "notes": "Notizen",
  "actions": "Aktionen"
}
