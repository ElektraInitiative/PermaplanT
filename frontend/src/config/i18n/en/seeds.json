{
  "view_seeds": {
    "title": "My Seeds",
    "btn_new_entry": "New Entry",
    "search_placeholder": "Search seeds...",
    "search_hint": "Currently, seed search only considers the additional name.",
    "fetching_error": "Sorry, I'm experiencing trouble fetching seeds from my server. Please try again later.",
    "fetching_single_error": "Sorry, I could not load the seed from my server. It is possible it doesn't exist.",
    "restore_seed_message": "Archived {{seed}}",
    "restore_seed_button": "Restore",
    "restore_seed_success": "Restored seed {{seed}} successfully!",
    "restore_seed_failure": "Could not restore seed.",
    "archive_seed_tooltip": "Archive seed",
    "edit_seed_tooltip": "Edit seed"
  },
  "create_seed": {
    "title": "Seed Entry",
    "changes_model_title": "Cancel Changes?",
    "changes_model_message": "Changes you have made will not be saved. Do you really want to cancel?"
  },
  "delete_seed": {
    "btn_delete_seed": "Delete seed"
  },
  "create_seed_form": {
    "placeholder_binomial_name": "Tomato",
    "placeholder_origin": "My Home",
    "placeholder_taste": "sweet",
    "placeholder_yield": "1",
    "placeholder_price": "2,99â‚¬",
    "placeholder_generation": "0",
    "error_price_must_be_number": "Price must be a number",
    "btn_create_seed": "Create Seed",
    "error_create_seed": "Sorry, I'm experiencing trouble storing the seed on my server. Please try again later.",
    "error_delete_seed": "Sorry, I'm experiencing trouble deleting the seed from the server. Please try again later.",
    "error_fetching_seeds": "Sorry, I'm experiencing trouble fetching seeds from my server. Please try again later.",
    "error_fetching_plants": "Sorry, I'm experiencing trouble fetching plants from my server. Please try again later.",
    "success": "Seed created successfully!"
  },
  "edit_seed_form": {
    "btn_edit_seed": "Edit seed",
    "success": "Seed modified successfully!"
  },
  "harvest_year": "Harvest Year",
  "additional_name": "Additional Name",
  "binomial_name": "Plant Name",
  "quantity": "Amount",
  "origin": "Origin",
  "use_by": "Best by",
  "quality": "Quality",
  "taste": "Taste",
  "yield": "Yield",
  "price": "Price",
  "generation": "Generation",
  "notes": "Notes",
  "actions": "Actions"
}
