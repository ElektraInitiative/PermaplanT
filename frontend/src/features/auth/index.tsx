export { LoginButton } from './components/LoginButton';
export { getAuthInfo } from './api/getAuthInfo';
