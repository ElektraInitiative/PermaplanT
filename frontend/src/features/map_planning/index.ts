export * from './routes/Map';
export * from './routes/CreateMap';
