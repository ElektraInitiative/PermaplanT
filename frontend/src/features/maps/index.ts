export * from './routes/MapOverview';
export * from './routes/MapCreateForm';
export * from './routes/MapEditForm';
