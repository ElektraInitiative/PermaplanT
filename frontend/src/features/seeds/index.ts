/**
 * The route exports from this module are not part of the public API.
 * @ignore
 * @module
 */
export * from './routes/CreateSeed';
export * from './routes/EditSeed';
export * from './routes/ViewSeeds';
