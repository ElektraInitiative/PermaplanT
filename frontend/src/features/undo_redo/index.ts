import useMapStore from './store/MapHistoryStore';

export * from './store/state-types';
export * from './store/action-types';
export default useMapStore;
