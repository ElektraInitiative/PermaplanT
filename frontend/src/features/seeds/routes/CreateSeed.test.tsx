import { CreateSeed } from '@/features/seeds';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const queryClient = new QueryClient();
  const tree = renderer
    .create(
      <QueryClientProvider client={queryClient}>
        <MemoryRouter>
          <CreateSeed></CreateSeed>
        </MemoryRouter>
      </QueryClientProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
