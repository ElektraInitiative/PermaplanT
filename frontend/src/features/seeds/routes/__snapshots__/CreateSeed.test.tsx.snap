// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mx-auto w-full px-4 py-8 md:w-[900px] undefined"
>
  <h1
    className="mb-8 text-2xl"
  >
    Seed Entry
  </h1>
  <div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="mb-6 grid gap-8 md:grid-cols-2"
      >
        <div
          data-testid="paginated-select-menu__Plant Name"
        >
          <label
            className="mb-2 block text-sm font-medium"
            htmlFor="plant_id"
          >
            Plant Name
            <span
              className="text-red-800"
            >
               *
            </span>
          </label>
          <div
            className=" css-b62m3t-container"
            onKeyDown={[Function]}
          >
            <span
              className="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
            />
            <div
              className="h-[44px] bg-neutral-200 rounded border
        dark:bg-neutral-50-dark focus:border-primary-500
        hover:border-primary-500 dark:focus:border-primary-300 dark:hover:border-primary-300
         dark:border-neutral-400-dark border-neutral-500 css-qz5xir-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className="flex-nowrap css-1fdsijx-ValueContainer"
              >
                <div
                  className=" css-1jqq78o-placeholder"
                  id="react-select-2-placeholder"
                >
                  Tomato
                </div>
                <div
                  className=" css-qbdosj-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-describedby="react-select-2-placeholder"
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-required={true}
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    className=""
                    disabled={false}
                    id="react-select-2-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    role="combobox"
                    spellCheck="false"
                    style={
                      {
                        "background": 0,
                        "border": 0,
                        "color": "inherit",
                        "font": "inherit",
                        "gridArea": "1 / 2",
                        "label": "input",
                        "margin": 0,
                        "minWidth": "2px",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "100%",
                      }
                    }
                    tabIndex={0}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-1xc3v61-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-tj5bde-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              aria-hidden="true"
              className="css-5kkxb2-requiredInput-RequiredInput"
              name="plant_id"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              tabIndex={-1}
              value=""
            />
          </div>
        </div>
        <div
          className="dark:text-white"
        >
          <label
            className="mb-2 block text-sm font-medium"
            htmlFor="name"
          >
            Additional Name
            <span
              className="text-red-800"
            >
               *
            </span>
          </label>
          <input
            aria-label="Additional Name"
            className="input border border-neutral-500 bg-neutral-100 placeholder-neutral-500 focus:border-primary-500 focus:outline-none disabled:cursor-not-allowed disabled:border-neutral-400 disabled:text-neutral-400 aria-[invalid=true]:border-red-400 dark:border-neutral-400-dark dark:bg-neutral-50-dark dark:focus:border-primary-300 dark:disabled:border-neutral-400-dark dark:disabled:text-neutral-400-dark dark:aria-[invalid=true]:border-red-400 "
            id="name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            pattern="^(?!.*(-))(?=.*[a-zA-Z0-9äöüÄÖÜß]).*$"
            placeholder=""
            required={true}
            title="At least one letter or number, '-' is not allowed"
          />
        </div>
        <div
          className="dark:text-white"
        >
          <label
            className="mb-2 block text-sm font-medium"
            htmlFor="harvest_year"
          >
            Harvest Year
            <span
              className="text-red-800"
            >
               *
            </span>
          </label>
          <input
            aria-label="Harvest Year"
            className="input border border-neutral-500 bg-neutral-100 placeholder-neutral-500 focus:border-primary-500 focus:outline-none disabled:cursor-not-allowed disabled:border-neutral-400 disabled:text-neutral-400 aria-[invalid=true]:border-red-400 dark:border-neutral-400-dark dark:bg-neutral-50-dark dark:focus:border-primary-300 dark:disabled:border-neutral-400-dark dark:disabled:text-neutral-400-dark dark:aria-[invalid=true]:border-red-400 "
            defaultValue={2023}
            id="harvest_year"
            name="harvest_year"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="2023"
            required={true}
            type="number"
          />
        </div>
        <div
          data-testid="Amount"
        >
          <label
            className="mb-2 block text-sm font-medium"
            htmlFor="quantity"
          >
            Amount
            <span
              className="text-red-800"
            >
               *
            </span>
          </label>
          <div
            className=" css-b62m3t-container"
            onKeyDown={[Function]}
          >
            <span
              className="css-1f43avz-a11yText-A11yText"
              id="react-select-3-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
            />
            <div
              className="h-[44px] bg-neutral-200 rounded border
        dark:bg-neutral-50-dark focus:border-primary-500
        hover:border-primary-500 dark:focus:border-primary-300 dark:hover:border-primary-300
         dark:border-neutral-400-dark border-neutral-500 css-qz5xir-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className="flex-nowrap css-1fdsijx-ValueContainer"
              >
                <div
                  className=" css-1g0ciz2-singleValue"
                >
                  More than enough
                </div>
                <div
                  className=" css-qbdosj-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-required={true}
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    className=""
                    disabled={false}
                    id="react-select-3-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    role="combobox"
                    spellCheck="false"
                    style={
                      {
                        "background": 0,
                        "border": 0,
                        "color": "inherit",
                        "font": "inherit",
                        "gridArea": "1 / 2",
                        "label": "input",
                        "margin": 0,
                        "minWidth": "2px",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "100%",
                      }
                    }
                    tabIndex={0}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-1xc3v61-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-tj5bde-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="quantity"
              type="hidden"
              value="more than enough"
            />
          </div>
        </div>
        <div
          className="dark:text-white"
        >
          <label
            className="mb-2 block text-sm font-medium"
            htmlFor="use_by"
          >
            Best by
          </label>
          <input
            className="input border border-neutral-500 bg-neutral-100 placeholder-neutral-500 focus:border-primary-500 focus:outline-none disabled:cursor-not-allowed disabled:border-neutral-400 disabled:text-neutral-400 aria-[invalid=true]:border-red-400 dark:border-neutral-400-dark dark:bg-neutral-50-dark dark:focus:border-primary-300 dark:disabled:border-neutral-400-dark dark:disabled:text-neutral-400-dark dark:aria-[invalid=true]:border-red-400 "
            id="use_by"
            name="use_by"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="date"
          />
        </div>
        <div
          className="dark:text-white"
        >
          <label
            className="mb-2 block text-sm font-medium"
            htmlFor="origin"
          >
            Origin
          </label>
          <input
            className="input border border-neutral-500 bg-neutral-100 placeholder-neutral-500 focus:border-primary-500 focus:outline-none disabled:cursor-not-allowed disabled:border-neutral-400 disabled:text-neutral-400 aria-[invalid=true]:border-red-400 dark:border-neutral-400-dark dark:bg-neutral-50-dark dark:focus:border-primary-300 dark:disabled:border-neutral-400-dark dark:disabled:text-neutral-400-dark dark:aria-[invalid=true]:border-red-400 "
            id="origin"
            name="origin"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="My Home"
          />
        </div>
        <div
          data-testid="Quality"
        >
          <label
            className="mb-2 block text-sm font-medium"
            htmlFor="quality"
          >
            Quality
          </label>
          <div
            className=" css-b62m3t-container"
            onKeyDown={[Function]}
          >
            <span
              className="css-1f43avz-a11yText-A11yText"
              id="react-select-4-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
            />
            <div
              className="h-[44px] bg-neutral-200 rounded border
        dark:bg-neutral-50-dark focus:border-primary-500
        hover:border-primary-500 dark:focus:border-primary-300 dark:hover:border-primary-300
         dark:border-neutral-400-dark border-neutral-500 css-qz5xir-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className="flex-nowrap css-1fdsijx-ValueContainer"
              >
                <div
                  className=" css-1g0ciz2-singleValue"
                >
                  Unknown
                </div>
                <div
                  className=" css-qbdosj-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-required={false}
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    className=""
                    disabled={false}
                    id="react-select-4-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    role="combobox"
                    spellCheck="false"
                    style={
                      {
                        "background": 0,
                        "border": 0,
                        "color": "inherit",
                        "font": "inherit",
                        "gridArea": "1 / 2",
                        "label": "input",
                        "margin": 0,
                        "minWidth": "2px",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "100%",
                      }
                    }
                    tabIndex={0}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-1xc3v61-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-tj5bde-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="quality"
              type="hidden"
              value="unknown"
            />
          </div>
        </div>
        <div
          className="dark:text-white"
        >
          <label
            className="mb-2 block text-sm font-medium"
            htmlFor="taste"
          >
            Taste
          </label>
          <input
            className="input border border-neutral-500 bg-neutral-100 placeholder-neutral-500 focus:border-primary-500 focus:outline-none disabled:cursor-not-allowed disabled:border-neutral-400 disabled:text-neutral-400 aria-[invalid=true]:border-red-400 dark:border-neutral-400-dark dark:bg-neutral-50-dark dark:focus:border-primary-300 dark:disabled:border-neutral-400-dark dark:disabled:text-neutral-400-dark dark:aria-[invalid=true]:border-red-400 "
            id="taste"
            name="taste"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="sweet"
          />
        </div>
        <div
          className="dark:text-white"
        >
          <label
            className="mb-2 block text-sm font-medium"
            htmlFor="yield_"
          >
            Yield
          </label>
          <input
            className="input border border-neutral-500 bg-neutral-100 placeholder-neutral-500 focus:border-primary-500 focus:outline-none disabled:cursor-not-allowed disabled:border-neutral-400 disabled:text-neutral-400 aria-[invalid=true]:border-red-400 dark:border-neutral-400-dark dark:bg-neutral-50-dark dark:focus:border-primary-300 dark:disabled:border-neutral-400-dark dark:disabled:text-neutral-400-dark dark:aria-[invalid=true]:border-red-400 "
            id="yield_"
            name="yield_"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="good"
          />
        </div>
        <div
          className="dark:text-white"
        >
          <label
            className="mb-2 block text-sm font-medium"
            htmlFor="price"
          >
            Price
          </label>
          <input
            className="input border border-neutral-500 bg-neutral-100 placeholder-neutral-500 focus:border-primary-500 focus:outline-none disabled:cursor-not-allowed disabled:border-neutral-400 disabled:text-neutral-400 aria-[invalid=true]:border-red-400 dark:border-neutral-400-dark dark:bg-neutral-50-dark dark:focus:border-primary-300 dark:disabled:border-neutral-400-dark dark:disabled:text-neutral-400-dark dark:aria-[invalid=true]:border-red-400 "
            id="price"
            max={327}
            min={0}
            name="price"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="2.99"
            step="0.01"
            title="Sorry, I only understand the price as a number with dot (.) or comma (,) depending on your Browser's preferences as decimal separator, e.g. 2.99"
            type="number"
          />
        </div>
        <div
          className="dark:text-white"
        >
          <label
            className="mb-2 block text-sm font-medium"
            htmlFor="generation"
          >
            Generation
          </label>
          <input
            className="input border border-neutral-500 bg-neutral-100 placeholder-neutral-500 focus:border-primary-500 focus:outline-none disabled:cursor-not-allowed disabled:border-neutral-400 disabled:text-neutral-400 aria-[invalid=true]:border-red-400 dark:border-neutral-400-dark dark:bg-neutral-50-dark dark:focus:border-primary-300 dark:disabled:border-neutral-400-dark dark:disabled:text-neutral-400-dark dark:aria-[invalid=true]:border-red-400 "
            id="generation"
            min={0}
            name="generation"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="0"
            type="number"
          />
        </div>
      </div>
      <div
        className="mb-6"
      >
        <div
          className="dark:text-white"
        >
          <label
            className="mb-2 block text-sm font-medium"
            htmlFor="notes"
          >
            Notes
          </label>
          <textarea
            className="input border border-neutral-500 bg-neutral-100 placeholder-neutral-500 focus:border-primary-500 focus:outline-none disabled:cursor-not-allowed disabled:border-neutral-400 disabled:text-neutral-400 aria-[invalid=true]:border-red-400 dark:border-neutral-400-dark dark:bg-neutral-50-dark dark:focus:border-primary-300 dark:disabled:border-neutral-400-dark dark:disabled:text-neutral-400-dark dark:aria-[invalid=true]:border-red-400 "
            id="notes"
            name="notes"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="..."
            type="textarea"
          />
        </div>
      </div>
      <div
        className="flex flex-row justify-between space-x-4"
      >
        <button
          className="button disabled:bg-neutral-300 disabled:border-neutral-300 disabled:text-neutral-500 dark:disabled:bg-neutral-300-dark dark:disabled:border-neutral-300-dark dark:disabled:text-neutral-500-dark disabled:cursor-not-allowed focus:outline-none focus:ring-4 focus:ring-blue-300 border bg-secondary-500 dark:bg-secondary-300 border-secondary-500 hover:bg-secondary-600 dark:hover:bg-secondary-200 active:bg-secondary-900 text-secondary-50 dark:text-secondary-700 max-w-[240px] grow sm:w-auto"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="button disabled:bg-neutral-300 disabled:border-neutral-300 disabled:text-neutral-500 dark:disabled:bg-neutral-300-dark dark:disabled:border-neutral-300-dark dark:disabled:text-neutral-500-dark disabled:cursor-not-allowed focus:outline-none focus:ring-4 focus:ring-blue-300 border bg-primary-500 dark:bg-primary-300 border-primary-500 hover:bg-primary-600 dark:hover:bg-primary-200 active:bg-primary-900 text-primary-50 dark:text-primary-700 max-w-[240px] grow sm:w-auto"
          title="Create Seed"
          type="submit"
        >
          Create Seed
        </button>
      </div>
    </form>
  </div>
  <div
    className="fixed bottom-0 right-0 z-[1010] pointer-events-none"
    onClick={[Function]}
  />
</div>
`;
